### PR TITLE
fix: lib/runtime: allow generic keystore type in public_keys ext functions

### LINKED ISSUE #1279: update runtime keystore to use GlobalKeystore, use key type IDs in ext_crypto funcs
## Task summary

- currently the runtime uses only the `Acco` keystore
- it should be using the `GlobalKeystore` instead (see dot/node.go `createRuntime`)
- the runtime should then use the correct keystore based on the `keyTypeID` that's passed into ext_crypto_... funcs
- the `keyTypeID` corresponds to the keystore's `Name` field

## Specification reference

- spec appendix E ext_crypto funcs

## Other information and links

-

### PR DESCRIPTION
**Pull request by noot**
## Changes

- allow generic keystore type in public_keys ext functions (since it may contain any type of key)

## Tests

```
go test ./lib/runtime/wasmer/ -run Test_ext_crypto_ed25519_public_keys_version_1
go test ./lib/runtime/wasmer/ -run Test_ext_crypto_sr25519_public_keys_version_1
```

## Checklist

- [x] I have read [CODE_OF_CONDUCT](https://github.com/ChainSafe/gossamer/blob/development/.github/CODE_OF_CONDUCT.md) and [CONTRIBUTING](https://github.com/ChainSafe/gossamer/blob/development/.github/CONTRIBUTING.md) 
- [x] I have provided as much information as possible and necessary
- [x] I have reviewed my own pull request before requesting a review
- [x] All integration tests and required coverage checks are passing

## Issues

- closes #1279

### CODE REVIEW

**Comment by FlorianFranzen** [comment]
Thank you, this fixes the issue for us!

**Review by FlorianFranzen — APPROVED** [review]

File: lib/runtime/wasmer/imports.go [hunk 1]
```diff
@@ -526,8 +526,8 @@ func ext_crypto_sr25519_public_keys_version_1(context unsafe.Pointer, keyTypeID
 		return C.int64_t(ret)
 	}
 
-	if ks.Type() != crypto.Sr25519Type {
-		logger.Warn("[ext_crypto_ed25519_public_keys_version_1]", "name", id, "error", "keystore type is not ed25519")
+	if ks.Type() != crypto.Sr25519Type && ks.Type() != crypto.UnknownType {
+		logger.Warn("[ext_crypto_sr25519_public_keys_version_1]", "name", id, "error", "keystore type is not ed25519")
```
Comment (RyRy79261) [inline lib/runtime/wasmer/imports.go:530]: Just need to update the error message for the correct key type

```suggestion
		logger.Warn("[ext_crypto_sr25519_public_keys_version_1]", "name", id, "error", "keystore type is not sr25519")
```

Comment (RyRy79261) [reply lib/runtime/wasmer/imports.go:530]: Or if this is just checking for er25519, wouldn't this log if the key type is ed25519?

**Review by RyRy79261 — DISMISSED** [review]

**Review by edwardmack — APPROVED** [review]

**Review by arijitAD — APPROVED** [review]
https://play.golang.org/p/B4-xirJsSKd
We can implement a custom logger for `wasmer` package that prints the function name dynamically.

**Review by RyRy79261 — APPROVED** [review]

**Pull request merged**